options/options_settable_test.cc

```diff
@@ -319,75 +322,61 @@ TEST_F(OptionsSettableTest, ColumnFamilyOptionsAllFieldsSettable) {
-      {offset_of(&ColumnFamilyOptions::comparator), sizeof(Comparator*)},
-      {offset_of(&ColumnFamilyOptions::merge_operator),
+      {offset_of(ColumnFamilyOptions, comparator), sizeof(Comparator*)},
+      {offset_of(ColumnFamilyOptions, merge_operator),
        sizeof(std::shared_ptr<MergeOperator>)},
-      {offset_of(&ColumnFamilyOptions::compaction_filter),
+      {offset_of(ColumnFamilyOptions, compaction_filter),
        sizeof(const CompactionFilter*)},
-      {offset_of(&ColumnFamilyOptions::compaction_filter_factory),
+      {offset_of(ColumnFamilyOptions, compaction_filter_factory),
        sizeof(std::shared_ptr<CompactionFilterFactory>)},
-      {offset_of(&ColumnFamilyOptions::prefix_extractor),
+      {offset_of(ColumnFamilyOptions, prefix_extractor),
        sizeof(std::shared_ptr<const SliceTransform>)},
-      {offset_of(&ColumnFamilyOptions::table_factory),
+      {offset_of(ColumnFamilyOptions, table_factory),
        sizeof(std::shared_ptr<TableFactory>)},
   };
 
   char* options_ptr = new char[sizeof(ColumnFamilyOptions)];
 
   // Count padding bytes by setting all bytes in the memory to a special char,
-  // copy a well constructed struct to this memory and see how many special
-  // bytes left.
-  ColumnFamilyOptions* options = new (options_ptr) ColumnFamilyOptions();
+  // then, default construct the object to make sure that everything
+  // initialized by default, then count the special chars
   FillWithSpecialChar(options_ptr, sizeof(ColumnFamilyOptions),
                       kColumnFamilyOptionsBlacklist);
-  // It based on the behavior of compiler that padding bytes are not changed
-  // when copying the struct. It's prone to failure when compiler behavior
-  // changes. We verify there is unset bytes to detect the case.
-  *options = ColumnFamilyOptions();
+  ColumnFamilyOptions* options = new (options_ptr) ColumnFamilyOptions();
 
-  // Deprecatd option which is not initialized. Need to set it to avoid
-  // Valgrind error
+  // Make sure this is inited
   options->max_mem_compaction_level = 0;
 
   int unset_bytes_base = NumUnsetBytes(options_ptr, sizeof(ColumnFamilyOptions),
                                        kColumnFamilyOptionsBlacklist);
   ASSERT_GT(unset_bytes_base, 0);
-  options->~ColumnFamilyOptions();
-
-  options = new (options_ptr) ColumnFamilyOptions();
-  FillWithSpecialChar(options_ptr, sizeof(ColumnFamilyOptions),
-                      kColumnFamilyOptionsBlacklist);
-
-  // Following options are not settable through
-  // GetColumnFamilyOptionsFromString():
-  options->rate_limit_delay_max_milliseconds = 33;
-  options->compaction_options_universal = CompactionOptionsUniversal();
-  options->compression_opts = CompressionOptions();
-  options->hard_rate_limit = 0;
-  options->soft_rate_limit = 0;
-  options->purge_redundant_kvs_while_flush = false;
-  options->max_mem_compaction_level = 0;
 
+  // Here we do the reverse, do a well constructed object
```

Comment:
unfortunately I think this means the code above for refilling `*options` with special chars is necessary. Otherwise we won't be able to ensure the `GetColumnFamilyOptionsFromString` is setting options via deserializing the options string, rather than via the copy-assignment I pointed out below.